source/frontend/src/features/documents/DocumentPreviewContainer.tsx

```diff
@@ -0,0 +1,167 @@
+import moment from 'moment';
+import { FC, useCallback, useContext, useEffect, useState } from 'react';
+
+import LoadingBackdrop from '@/components/common/LoadingBackdrop';
+import { ModalContext } from '@/contexts/modalContext';
+import { ApiGen_Mayan_FilePage } from '@/models/api/generated/ApiGen_Mayan_FilePage';
+import { ApiGen_System_HttpStatusCode } from '@/models/api/generated/ApiGen_System_HttpStatusCode';
+
+import { DocumentViewerContext } from './context/DocumentViewerContext';
+import { IDocumentPreviewViewProps } from './DocumentPreviewView';
+import { createFileDownload } from './DownloadDocumentButton';
+import { useDocumentProvider } from './hooks/useDocumentProvider';
+
+export interface LoadedPages extends ApiGen_Mayan_FilePage {
```

Comment:
- Singular (LoadedPage)
- It would be better if this uses composition instead of inheritance given a loaded page contains both Mayan page information as well as the image retrieved for that page. That would help disambiguate things once that data is used somewherelese